libc/Makefile

```diff
@@ -17,6 +17,19 @@ LOCAL_LDFLAGS := -z stack-size=12288 $(2)
 include $(binary.mk)
 endef
 
+# Add libc test using custom sources from subdir
+# $(eval $(call add_libc_test,SUBDIR_NAME,NAME[,LOCAL_LDFLAGS][,LOCAL_CFLAGS]))
+define add_test_libc_custom
+NAME := test-libc-$(2)
+LIBC_UNIT_TESTS += test-libc-$(2)
+SRCS := $(addprefix $(MY_LOCAL_DIR)$(1)/, $(CUSTOM_SRCS)) $(COMMON_SRCS)
+DEP_LIBS := unity
+LOCAL_CFLAGS := -I$(MY_LOCAL_DIR) $(4)
+LOCAL_LDFLAGS := -z stack-size=12288 $(3)
+
+include $(binary.mk)
+endef
+
```

Comment:
I don't like using almost the same macro again. I get that you want to make two different tests from one folder because not all tests will work on every target. But, can't we just use an if statement in the macro for setting SRCS, or split the tests into two folders instead?